src/subtype.c

```diff
@@ -191,9 +191,9 @@ static int obviously_unequal(jl_value_t *a, jl_value_t *b)
 {
     if (a == b)
         return 0;
-    if (jl_is_leaf_type(a) && !((jl_datatype_t*)a)->abstract)
+    if (a != jl_bottom_type && jl_is_leaf_type(a) && !((jl_datatype_t*)a)->abstract)
```

Comment:
a better option here would be `a == jl_bottom_type || (jl_is_leaf_type(a) && !abstract)`